### ISSUE TITLE
Problem starting molcalc server: `configparser.NoSectionError: No section: 'molcalc'`

### ISSUE THREAD

**Issue by mscloudlab** [jensengroup/molcalc#36]
We've been trying to get a molcalc server running on one of our workstations by following the instructions in the readme.rst. However, we're currently stuck on Step 7. After running

```
$ ./env/bin/pserve development.ini --reload
```

we get the following Trackback

```
Starting monitor for PID 1576308.
Traceback (most recent call last):
  File "/home/cloudlab/Repositories/python/molcalc/env/lib/python3.7/configparser.py", line 845, in items
    d.update(self._sections[section])
KeyError: 'molcalc'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/cloudlab/Repositories/python/molcalc/env/lib/python3.7/site-packages/hupper/ipc.py", line 319, in spawn_main
    func(**kwargs)
  File "/home/cloudlab/Repositories/python/molcalc/env/lib/python3.7/site-packages/hupper/worker.py", line 252, in worker_main
    func(*spec_args, **spec_kwargs)
  File "/home/cloudlab/Repositories/python/molcalc/env/lib/python3.7/site-packages/pyramid/scripts/pserve.py", line 30, in main
    return command.run()
  File "/home/cloudlab/Repositories/python/molcalc/env/lib/python3.7/site-packages/pyramid/scripts/pserve.py", line 271, in run
    app = loader.get_wsgi_app(app_name, config_vars)
  File "/home/cloudlab/Repositories/python/molcalc/env/lib/python3.7/site-packages/plaster_pastedeploy/__init__.py", line 129, in get_wsgi_app
    global_conf=defaults,
  File "/home/cloudlab/Repositories/python/molcalc/env/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 253, in loadapp
    return loadobj(APP, uri, name=name, **kw)
  File "/home/cloudlab/Repositories/python/molcalc/env/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 278, in loadobj
    return context.create()
  File "/home/cloudlab/Repositories/python/molcalc/env/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 715, in create
    return self.object_type.invoke(self)
  File "/home/cloudlab/Repositories/python/molcalc/env/lib/python3.7/site-packages/paste/deploy/loadwsgi.py", line 152, in invoke
    return fix_call(context.object, context.global_conf, **context.local_conf)
  File "/home/cloudlab/Repositories/python/molcalc/env/lib/python3.7/site-packages/paste/deploy/util.py", line 55, in fix_call
    val = callable(*args, **kw)
  File "/home/cloudlab/Repositories/python/molcalc/molcalc/__init__.py", line 135, in main
    for k, v in parser.items(section):
  File "/home/cloudlab/Repositories/python/molcalc/env/lib/python3.7/configparser.py", line 848, in items
    raise NoSectionError(section)
configparser.NoSectionError: No section: 'molcalc'
Press ENTER or change a file to reload.
```

The error appears to be that the *.ini file needs a "molcalc" section, but we don't have enough context to guess what exactly is going on and what is needed.

We're very happy to provide as much information as we can to help, including details about the steps leading up to Step 7.

Thanks in advance!

**Comment by charnley** [comment]
Hi mscloudlab, sorry, always a challenge to keep all projects up to date. It is not used for anything atm. It was supposed to be used to block IP's, but was never implemented.

> [molcalc]
> block_ip=

I pushed a small bug fix to ignore sections that don't exist.

Another note, when using ppqm, please don't use the newest version. Please check out ppqm commit `f36da9a` for molcalc. Currently, there is no versioning of the ppqm package.

I also see that ChemDoodleWeb 8.0.0 is not available anymore, but getting the newest should also work.

I'll keep this issue open in case there are other things for a clean setup. Let me know when you have a version up and running and what you need to do (differently from readme) to make it work.

**Comment by charnley** [comment]
Thanks! Good work getting GAMESS working :) 

I don't remember exactly what was up with that unittest, but that will be left for the future. Will do some refactor when ppqm is published in conda, and make sure molcalc is future deploy safe.

**Comment by mscloudlab** [comment]
Aside from working with commit `f36da9a`, here's a quick summary of the steps taken to get a molcalc server going:

1. Same as Step 1 in `readme.rst`.
2. Same as Step 2 in `readme.rst`, but the symbolic link was changed to point to `ppqm.git/ppqm` (rather than just `ppqm/ppqm`), in accordance with "`git clone https://github.com/ppqm/ppqm ppqm.git --depth 1`." I.e.,
```python
$ ln -s ppqm.git/ppqm ppqm
```
3. Same commands as Step 3 in `readme.rst`, but `scripts/setup_chemdoodle.sh` was just modified with a variable to accept different version numbers of ChemDoodleWeb and then the version was set to the most recent version (v9.4.0):[setup_chemdoodle.sh.txt](https://github.com/jensengroup/molcalc/files/8921823/setup_chemdoodle.sh.txt). Maybe the version number can just be made a command-line argument? Also, perhaps a link to [ChemDoodle Web Components download page](https://web.chemdoodle.com/installation/download) would also be helpful (so users don't confuse the web components with the downloads ChemDoodle 2D/3D from the [ChemDoodle site](https://www.chemdoodle.com/) in the event a user chooses to look at ChemDoodleWeb themselves).
4. Same (hell) as Step 4  in `readme.rst`. We actually had to compile GAMESS with `ifort` to get it to work (using Intel's oneAPI package for Linux). (There always seem to be bizarre problems that crop up when compiling with `gfortran`, especially with codes that mix old/new Fortran standards.) Perhaps it's also worth noting that we compiled GAMESS with none of the extra packages, selecting "sockets" rather than MPI, and skipping the use of any of the accelerated math libraries.
5. Same as Step 5 in `readme.rst`.
6. Close to Step 6 in `readme.rst`, but installed the dev requirements too so pytest could be run. (There might have been an additional package that had to also be installed manually, but it was simple to do via `conda install`.)
```
pip install -r requirements.dev.txt
python -m pytest tests
```
Importantly, however, was that the SMILES test for the sodium cation failed, as mentioned before. This was simply deleted from the list assigned to the variable `SMALL_SMILES` in `test_chemistry.py`.
7. Same as Step 7 in `readme.rst` once the most recent bug fixes we're merged.

Hope this is helpful, and we'd be happy to provide any other information that we can.
